### ISSUE TITLE
docs need clear example of how to cancel/abort

### ISSUE THREAD

**Issue by mspanish** [Mitranim/xhttp#16]
I see the section in the secondary api about cancellation - but could you add a clear example of how to cancel a request? For example, my request is found inside a loop and looks like this:

```
xhr = Xhttp(params, ({ok, status, reason, headers, body}) => {
	if (!ok) console.warn('Failure:', body)
// all of my processing code here
})
```

**Comment by mspanish** [comment]
Another issue is if I'm using this in a loop (maybe that's be in and of itself!) - I assume I'll need to save a ref to each xhr in order to cancel that specific request, right? Could I push these into an array, then loop back through to cancel each individual xhr?

**Comment by Mitranim** [comment]
Added cancelation instructions to the readme: https://github.com/Mitranim/xhttp/tree/18051e866a2c19efda5bac306b8469fa8b8f401a#cancelation

**Comment by Mitranim** [comment]
Yeah to track and possibly abort all requests, you need to store them. Make sure to remove them when done, or they'll accumulate indefinitely.

As an aside, since `XMLHttpRequest` and, by extension, this library, are callback-based, coordinating them can get annoying real quick. For example, if you have 12 requests and want to wait until they _all_ finish. That's what promises/futures are for:

```js
Promise.all([fetch('/one'), fetch('/two')]).then(/* ... */)
```

Of course promises don't support cancelation, so it's a nice way to "hang" your app when a user triggers dozens of now-unnecessary requests. That's the motivation for Posterus, which you've seen already. I've added a readme section suggesting how to use it with `xhttp`.
